src/xunit.analyzers/AssertEqualShouldNotBeUsedForCollectionSizeCheck.cs

```diff
@@ -68,33 +69,38 @@ protected override void Analyze(SyntaxNodeAnalysisContext context, InvocationExp
 
         private static bool IsWellKnownSizeMethod(SymbolInfo symbolInfo)
         {
-            return SizeMethods.Contains(SymbolDisplay.ToDisplayString(symbolInfo.Symbol.OriginalDefinition));
+            return SizeMethods.Contains(symbolInfo.Symbol.OriginalDefinition.ToDisplayString());
         }
 
-        private static bool IsCollectionCountProperty(SyntaxNodeAnalysisContext context, SymbolInfo symbolInfo)
+        private static bool IsICollectionCountProperty(SyntaxNodeAnalysisContext context, SymbolInfo symbolInfo)
         {
-            var collectionCountSymbol = context.Compilation
-                .GetTypeByMetadataName(typeof(ICollection).FullName)
-                .GetMembers(nameof(ICollection.Count))
-                .Single();
-
-            var collectionSymbolImplementation = symbolInfo.Symbol.ContainingType.FindImplementationForInterfaceMember(collectionCountSymbol);
-            return collectionSymbolImplementation != null && collectionSymbolImplementation.Equals(symbolInfo.Symbol);
+            var iCollection = context.Compilation.GetTypeByMetadataName(Constants.Types.SystemCollectionsICollection);
+            return IsCountPropertyOf(iCollection, symbolInfo);
         }
 
-        private static bool IsGenericCountProperty(SyntaxNodeAnalysisContext context, SymbolInfo symbolInfo)
+        private static bool IsICollectionOfTCountProperty(SyntaxNodeAnalysisContext context, SymbolInfo symbolInfo)
         {
-            if (symbolInfo.Symbol.ContainingType.TypeArguments.IsEmpty)
-                return false;
+            var containingType = symbolInfo.Symbol.ContainingType;
+            var openICollectionOfT = context.Compilation.GetSpecialType(SpecialType.System_Collections_Generic_ICollection_T);
+            var concreteICollectionOfT = containingType.GetGenericInterfaceImplementation(openICollectionOfT);
+            return concreteICollectionOfT != null && IsCountPropertyOf(concreteICollectionOfT, symbolInfo);
+        }
 
-            var genericCollectionCountSymbol = context.Compilation
-                .GetSpecialType(SpecialType.System_Collections_Generic_ICollection_T)
-                .Construct(symbolInfo.Symbol.ContainingType.TypeArguments.ToArray())
-                .GetMembers(nameof(ICollection<int>.Count))
-                .Single();
+        private static bool IsIReadOnlyCollectionOfTCountProperty(SyntaxNodeAnalysisContext context, SymbolInfo symbolInfo)
```

Comment:
This method is almost identical to `IsICollectionOfTCountProperty`, other than a different value of the `SpecialType` enum. Could be refactored into single implementation.